pyomo/gdp/disjunct.py

```diff
@@ -85,17 +85,21 @@ def transformation_block(self):
     def __init__(self, component):
         _BlockData.__init__(self, component)
         self.indicator_var = Var(within=Binary)
+        self.indicator_bool = BooleanVar()
```

Comment:
Also, Are we sure the BooleanVar and binary var in the disjunct will always be "in sync"?

Reply:
@qtothec, oops, I missed when you added this... Adding to Carl's questions, I'm a bit worried about naming here, and also having to maintain this API in the future. At some point we probably have to break backwards compatibility to really integrate this, but the problem is that, until then, we are establishing that users can assume that, *before transformation* a `Disjunct` has both `indicator_var` and `indicator_bool`. If we ever go to only a Boolean, now we have to choose one of those names and make it disappear altogether. I was assuming that some not-backwards-compatible day in the future, a `Disjunct` would be born with just one indicator_var, a BooleanVar. It would be the transformations' job to create the binary. If that's the case, then I'm not sure this is the right intermediate step? Or if I'm wrong, the broader question is: are we baking in any assumptions we don't want about what is inherent to a `Disjunct` and what is created by various transformations?

Last, if keeping both is what we want, should the BooleanVar be called `indicator_boolean` rather than `indicator_bool`, since I don't think we've abbreviated anywhere else?

Reply:
Yeah, I wanted to have the indicator for Disjuncts be BooleanVar, but I didn't want to immediately break backwards compatibility for everyone still using `indicator_var`.

I'm fine with having it named `indicator_boolean`.

Reply:
I think the right answer is to immediately move to "the new world order" where `indicator_var` are Boolean (not binary) variables.  I think I can create a "deprecation object" that derives from BooleanVar and mocks up the binary interface, so that if someone treats the `indicator_var` as a binary, it will toss a deprecation warning, and then "do the right thing" (probably return a binary, and ensure that the binary is associated with the Boolean).